pkg/k8s/apis/cilium.io/client/register.go

```diff
@@ -90,39 +90,121 @@ var (
 	comparableCRDSchemaVersion = versioncheck.MustVersion(k8sconst.CustomResourceDefinitionSchemaVersion)
 )
 
-type crdCreationFn func(clientset apiextensionsclient.Interface) error
+type CRDList struct {
+	ResourceName string
+	Name         string
+	FullName     string
+}
+
+// Returns a map of CRDs
+func CustomResourceDefinitionList() []*CRDList {
+	return []*CRDList{
```

Comment:
Couldn't we keep the CRDs in a map (i.e. `resourceToCreateFnMapping`) instead of converting it to a slice and use that in `CreateCustomResourceDefinitions`? Otherwise lookup in that func becomes `O(n^2)` instead of `O(n*log(n))`, i.e. for each entry of `synced.AllCiliumCRDResourceNames()` we'd search this list linearly.

For use in `tools/crdlistgen` we could instead lift out `resourceToCreateFnMapping` of `CreateCustomResourceDefinitions` and then let this func convert the map to a slice and sort it?

Reply:
I was trying to not create another slice somewhere that had to be maintained, and using the resourceToCreateFnMapping indexes would result in a fully lowercase list of CRDs so wouldn't be doc friendly.

Reply:
I see. How about introducing `CustomResourceDefinitionMap` that would keep this info in a map and that. `CreateCustomResourceDefinitions` would use and then let`CustomResourceDefinitionList` (which AFAICT is only used in `tools/crdlistgen`) call `CustomResourceDefinitionMap`, convert it to a slice and sort it?

Reply:
I have converted it back to a map. The tool doesn't need it as a slice, I just need access to the `Name` field for the docs which contains the "pretty" format of it.